### PR TITLE
Separate unit test configuration from the ones used in the README.

### PR DESCRIPTION
**Pull request by vpn-soft**
https://github.com/tetratelabs/istio-security-analyzer/issues/11

### CODE REVIEW

File: pkg/parser/parser_test.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 func loadTestConfigs(files ...string) ([]*istioConfig.Config, error) {
 	configObjects := make([]*istioConfig.Config, 0)
 	for _, inputTmpl := range files {
-		yamlBytes, err := ioutil.ReadFile("testdata/" + inputTmpl)
+		yamlBytes, err := ioutil.ReadFile("../../samples/" + inputTmpl)
```
Comment (incfly) [inline pkg/parser/parser_test.go:30]: hmm, I meant that for unit test we still have YAMLs in pkg/parser/testdata/ folder. So

- We have samples/ folder, contains everything except the "admingateway.yaml", "vs-deny-relaxed-sni.yaml" and "gw-simple-tls.yaml.
- We still have pkg/parser/testdata/, everything included.
- unit test still reference pkg/parser/testdata.

See the issue's explaining

> Additionally in some recent PR, we are adding unit test specific YAML config with positive and negative tests. They are not suitable and shouldn't be exposed to customers as well.

Because in our READEM we ask users to do `kubectl apply -f ./samples`. This should not contain more unit test specific config. We could have a sub-folder for 'gateway-tls-sni/', with additional document. but the idea is to separate the `samples/` structure from internal config in 'testdata/`.

Comment (jcchavezs) [reply pkg/parser/parser_test.go:30]: Agree, samples should focus on users, data for unit tests should focus on tests.

**Review by incfly — APPROVED** [review]
